### PR TITLE
[Merged by Bors] - feat: port Archive.Imo.Imo2008Q2

### PR DESCRIPTION
**Pull request by int-y1**
---

todo:

* [ ] `field_simp` doesn't understand `*`

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Archive/Imo/Imo2008Q2.lean [hunk 1]
```diff
@@ -37,10 +37,11 @@ namespace Imo2008Q2
 theorem subst_abc {x y z : ℝ} (h : x * y * z = 1) :
     ∃ a b c : ℝ, a ≠ 0 ∧ b ≠ 0 ∧ c ≠ 0 ∧ x = a / b ∧ y = b / c ∧ z = c / a := by
   use x, 1, 1 / y
-  obtain ⟨⟨hx, hy⟩, hz⟩ : (x ≠ 0 ∧ y ≠ 0) ∧ z ≠ 0 := by
-    simpa [not_or] using trans_rel_right (· ≠ ·) h one_ne_zero
+  obtain ⟨⟨hx, hy⟩, _⟩ : (x ≠ 0 ∧ y ≠ 0) ∧ z ≠ 0 := by
+    have := h.symm ▸ one_ne_zero
+    simpa [not_or] using this
```
Comment (semorrison) [inline Archive/Imo/Imo2008Q2.lean:42]: Is this a regression?

File: Archive/Imo/Imo2008Q2.lean [hunk 2]
```diff
@@ -37,10 +37,11 @@ namespace Imo2008Q2
...
   have : z * (y * x) = 1 := by rw [← h]; ac_rfl
-  field_simp [*]
+  field_simp [hx, this]
```
Comment (semorrison) [inline Archive/Imo/Imo2008Q2.lean]: Does `field_simp` not work with `[*]` anymore? If so, we should fix that.

File: Archive/Imo/Imo2008Q2.lean (hunk 1 above)
Comment (int-y1) [reply Archive/Imo/Imo2008Q2.lean:42]: I was getting rid of the deprecated function `trans_rel_right`

File: Archive/Imo/Imo2008Q2.lean (hunk 2 above)
Comment (int-y1) [reply Archive/Imo/Imo2008Q2.lean]: `field_simp [*]` didn't work here

Comment (jcommelin) [reply Archive/Imo/Imo2008Q2.lean]: Recorded as issue: #5689

This shouldn't block merging.

File: Archive/Imo/Imo2008Q2.lean [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+namespace Imo2008Q2
+
+theorem subst_abc {x y z : ℝ} (h : x * y * z = 1) :
+    ∃ a b c : ℝ, a ≠ 0 ∧ b ≠ 0 ∧ c ≠ 0 ∧ x = a / b ∧ y = b / c ∧ z = c / a := by
+  use x, 1, 1 / y
+  obtain ⟨⟨hx, hy⟩, _⟩ : (x ≠ 0 ∧ y ≠ 0) ∧ z ≠ 0 := by
+    have := h.symm ▸ one_ne_zero
+    simpa [not_or] using this
+  have : z * (y * x) = 1 := by rw [← h]; ac_rfl
+  field_simp [hx, this] -- Porting note: Originally `field_simp [*]`, but `*` doesn't work
```
Comment (jcommelin) [inline Archive/Imo/Imo2008Q2.lean:44]:
```suggestion
  -- Porting note: Originally `field_simp [*]`, but `*` doesn't work (see #5689)
  field_simp [hx, this]
```

**Comment by jcommelin** [comment]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Pull request closed without merging**